pages/_app.jsx

```diff
@@ -1,7 +1,50 @@
+import { useEffect, useState } from "react";
 import "@/styles/globals.scss";
 import Header from "@/components/Header";
 
 export default function App({ Component, pageProps }) {
+	const [theme, setTheme] = useState("bright");
+	const [fontSize, setFontSize] = useState(16);
```

Comment:
mentioned this in another comment but you shouldn't use a state here for the font size as you only use the local storage or the default value.

This is apparent if you look at the current chain of event:
first render: set font state ---effect---> second render: set font css variable 
there's no need for 2 steps as you can just do the second one

Reply:
changed?

Reply:
no? the state is still in `_app.jsx`. What was meant is that the state should be in the font size component and here you only listen get the value from local storage. In the component you change in runtime the value in css via useEffect

Reply:
noted